### DISCUSSION TITLE
ベータ版で実装する機能についてのメモ

### DISCUSSION THREAD

**Discussion by kazzyfrog** [lgtm-factory/lgtm-factory#76] [Scrap]
開発３周目にて、最低限の機能に絞って、実装します！

なので、下記を中心に、つらつら書いていきますー
- 最低限の機能の確認
- イシュー化するにあたっての実装の方向性？について

**Comment by kazzyfrog** [comment]
現状、必要なのは、こんな感じかな、、：
- 画像のURL コピーボタンの作成（GitHubマークダウン形式）
- 画像のダウンロードボタン
- 画像を生成するAPI
    - １つのエンドポイント
    - URLのバリデーションは別で。
    - キャッシュ周りの設定は別で。（完全に同じURLの場合、全く同じ画像が返却させるので、キャッシュしたい！）
    - エラーハンドリングも別でいいかも。（エラー時には、エラー用の画像返せたら良さそう。。！）
- 編集機能？の場合は、入力フォーム
- あとは、テーマ一覧の取得→画面に表示？
- そして、画像のデザインを複数用意・追加

**Reply by kazzyfrog** [reply]
もしかしたら、カスタマイズできる動的なパラメータごとの、
ざっくりしたまとまりで（カテゴリ毎）で、APIエンドポイント分けてもいいかも！

まあ、おいおいでOK！

**Reply by kazzyfrog** [reply]
あとは、編集機能がある場合は、
クライアントサイドで、生成する？毎回APIを叩く？など

**Comment by kazzyfrog** [comment]
デザイン一覧を取得して、表示するのも、どうしたもんか

**Comment by kazzyfrog** [comment]
APIをどうしたもんか。。

1. デザインの種類ごとに、APIエンドポイントを作成する：手動で（a/route.tsx, b/route.tsx,,,）
1. 動的なAPIエンドポイントを作成する：１つのファイルで（[design]/route.tsx）
1. １つのAPIエンドポイント：（api/route.tsx）

んーーー🧐

**Reply by kazzyfrog** [reply]
> １つのAPIエンドポイント：（api/route.tsx）

やっぱり、これで全然いける。。！

**Comment by kagomen** [comment]
<img width="1271" alt="スクリーンショット 2024-08-20 0 21 02" src="https://github.com/user-attachments/assets/c6ed745f-1ec2-4f99-8c77-deccd551bc23">


https://satori-practice-2.vercel.app
https://github.com/kagomen/satori-practice-2

**Reply by kagomen** [reply]
> 1. デザインの種類ごとに、APIエンドポイントを作成する

の方法で、デモページ作ってみました
実装の手軽さはこの方法が一番ですかね・・・

**Reply by kazzyfrog** [reply]
すごい。。！！

shadcn/uiも使って、横から出てくるのいいですね！👍
もうこのスレ内の、βの実装完了ですね！w

あと、実装も、実際に画面上で見れるのイメージ沸きますね！
参考になります！！

**Reply by kazzyfrog** [reply]
> 1. デザインの種類ごとに、APIエンドポイントを作成する

この方法だと、デザインが２０とか、３０とかになると大変かもですよね！
あと、メンテナー以外のコントリビュータ〜がデザイン作成したい場合、APIの処理まで書いてもらうのかとか、、、ってなると思うので！

結局、動的なのって、
- JSXエレメント
- オプション

だけなので、V1の本リリース？では、
上手に抽象化したいなーと思っているところです！😎

**Reply by kagomen** [reply]
いえ！！！全然です
CSSとかひどすぎて・・・（Discordにも書いたんですが、gridが必要かも？）

フロントもバックも、頭の中で構築するの難しすぎます・・・！！！🙀🙀

APIの設計の全種類作ってみようと思ったんですが、

- 動的なAPIエンドポイントを作成する：１つのファイルで（[design]/route.tsx）
- １つのAPIエンドポイント：（api/route.tsx）

↑これらが少し考えてみようとすると、頭爆発しそうになり……
大人しく、kazzyfrogさんの設計（？）を待とう、と思いました

**Reply by kagomen** [reply]
> この方法だと、デザインが２０とか、３０とかになると大変かもですよね！
あと、メンテナー以外のコントリビュータ〜がデザイン作成したい場合、APIの処理まで書いてもらうのかとか、、、ってなると思うので！

そうなんですよね・・・！

> 上手に抽象化したいなーと思っているところです

お願いします・・・・！🙏
カスタムフォントいれるとなると、難しそう・・・！でも、あるとすごく見栄えが良くなる・・・

**Reply by kazzyfrog** [reply]
> gridが必要かも？

確かにですね！！

なんとなく、いけた感じなので、明日イシュー立てます！！👍

**Reply by kazzyfrog** [reply]
めっちゃ雑ですが、こんな感じ。。かな
api: src/app/api/generate-image/route.tsx
https://github.com/lgtm-factory/lgtm-factory/blob/_playground/src/app/api/generate-image/route.tsx

デザインデータ：jsx-data
https://github.com/lgtm-factory/lgtm-factory/tree/_playground/jsx-data

**Reply by kagomen** [reply]
わー！早い・・・！ありがとうございます🙏🙏
ソースコード今日中に確認します！！！

> 明日イシュー立てます

感謝・・・！！！🙇🙇
